### ISSUE TITLE
`@code_typed optimize=true not_a_call` fails to throw a meaningful error

### ISSUE THREAD

**Issue by tkf** [JuliaLang/julia#34179]
```julia
julia> @code_typed optimize=true not_a_call
ERROR: function error does not accept keyword arguments
Stacktrace:
 [1] kwfunc(::Any) at ./boot.jl:332
 [2] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/InteractiveUtils/src/macros.jl:143

julia> @macroexpand @code_typed optimize=true no_a_call
quote
    #= /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/InteractiveUtils/src/macros.jl:143 =#
    var"#139#results" = InteractiveUtils.error("expression is not a function call or symbol", optimize=true)
    #= /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/InteractiveUtils/src/macros.jl:144 =#
    if InteractiveUtils.length(var"#139#results") == 1
        var"#139#results"[1]
    else
        var"#139#results"
    end
end
```

Note the keyword argument `optimize=true` in `InteractiveUtils.error("...", optimize=true)`.

This happens with 1.3 and 1.4.0-DEV.634.

**Comment by Liozou** [comment]
Incidentally fixed by #35522 I believe? It now appears like
```julia
julia> @code_typed optimize=true not_a_call
ERROR: expression is not a function call or symbol
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at /home/liozou/julia/usr/share/julia/stdlib/v1.6/InteractiveUtils/src/macros.jl:220
```

**Comment by tkf** [comment]
Yes, it works now. Thanks!
